lcls-twincat-pmps/PMPS/HelperFunctions/F_SetBeamParams.TcPOU

```diff
@@ -15,7 +15,7 @@ VAR
 END_VAR
 ]]></Declaration>
     <Implementation>
-      <ST><![CDATA[BeamParams.nTran := LIMIT(0,nTran,100);
+      <ST><![CDATA[BeamParams.nTran := LIMIT(0,nTran,PMPS_GVL.TRANS_SCALING_FACTOR);
```

Comment:
Does this mean that all attenuators and consumers of attenuation need to report on and check attenuation relative to this global variable?

Reply:
That sounds alarming.
My reasoning was that this constant would be available to remind us/ assist us with handling the fixed point representation of transmission, but it would not be required per-se to do anything with it.
Thoughts?

Reply:
I agree that we should not have do special handling in the other PLCs- I'm asking if this code makes it so that we do. That's what it looks like to me, `BeamParams` is expected to be maybe as high as `PMPS_GVL.TRANS_SCALING_FACTOR`

Reply:
You're right. It's ugly.

Two things I would offer to address this point.

1. This arrangement has precedent. We already use a global constant for the eV bitmask.
2. I think an overhaul of the BP struct is in-order. The only reason I started with UINT16 here is lack of space in the EL6692 data exchange PDO (480b, current bp io struct is at 148b). We can fix that in two ways. Update everyone to an EL6695, or remodel the BP io structure (painful). This needs to happen soon anyways, maybe just not right now. When we go to update the BP IO struct, we can change these attenuator values to REAL32 and cut the number of attenuators in the array in half and we'll be OK.

As a bonus, I can add something to the arbiter codebase to make constants like this available to the general PLC public through CA. PMPS lib PLCs could have an FB that checks important constants against the Arbiter.

Reply:
The global constant for the eV bitmask has run us into spots of trouble here and there, particularly when it changes. So I'm not sure that's very good precedent.

I'm :+1: on updating everything to EL6695 as you originally intended.